src/useObserver.ts

```diff
@@ -32,24 +33,27 @@ export function useObserver<T>(fn: () => T, baseComponentName: string = "observe
         // reaction from an abandoned render was disposed).
 
         const newReaction = new Reaction(observerComponentNameFor(baseComponentName), () => {
+            if (!reactionTrackingRef.current) {
+                // this branch is not expected to happen
```

Comment:
Maybe elaborate more on why this can never happen? I mean this whole block is into the same condition. Isn't it assumption based on the implementation of Reaction? What if at some point in the future it will be necessary to run that callback sooner?

Reply:
I'm not sure if i have a good one,

Maybe I will restore the old code here, and pass the whole ref to addReactionToTrack

Reply:
Changes pushed